doc/dart_documentation_comment_specification.md

```diff
@@ -0,0 +1,478 @@
+# **Dart Documentation Comments Specification**
+Version: 1.0.0
+
+[TOC]
+
+## **1\. Introduction**
+
+### **1.1. Purpose**
+
+The purpose of this document is to provide a single specification for documentation comments in the Dart programming language. Its goal is to ensure that documentation is parsed consistently by all tools and to serve as a definitive standard for developers and tool authors.
+
+### **1.2. Scope**
+
+**In Scope**
+
+This specification covers the following topics:
+
+* The *syntax* for writing documentation comments.
+* The rules governing where documentation comments can be *placed*.
+* The mechanism for *resolving* references between elements.
+* The logic for *associating* documentation with code elements.
+
+**Out of Scope**
+
+This document does not cover:
+
+* The implementation details of specific documentation tools (e.g., `dartdoc`, `analysis server`).
+* The syntax and behavior of tool-specific directives (e.g., `@template`, `@canonicalFor`).
+* Any details related to the visual styling or HTML layout of generated documentation.
+* Best practices or style guides for writing effective documentation content.
+
+### **1.3. Terminology**
+
+* **Doc Comment:** A comment intended to be processed by documentation generation tools, like  dartdoc.
+* **Element**: A specific, declared element in the Dart code, such as a class, method, function, variable, or type parameter.
+* **Identifier:** An individual name in the code (e.g., `MyClass`, myMethod, prefix).
+* **Qualified Name:** A name composed of two or more *identifiers* separated by dots, used to access an element within a specific namespace (e.g., MyClass.myMethod, prefix.MyClass).
+* **Reference:** A string enclosed in square brackets within a doc comment (e.g., `[MyClass]` or `[MyClass.myMethod]`) that links to an element. The text inside the brackets is either a single *identifier* or a *qualified name*.
+
+## **2\. Syntax of Documentation Comments**
+
+### **2.1. Line-Based Doc Comments (Recommended standard)**
+
+* Starts with `///`.
+* All consecutive lines starting with `///` are treated as part of the same comment block.
+
+### **2.2. Block-Based Doc Comments (Historic, not recommended)**
+
+* Starts with `/**` and ends with the matching `*/`.
+* Block comments can be nested (e.g., `\*\* \* inner */ */`). Documentation tools must respect this, ensuring that the comment block only ends at the closing  \*/ that matches the opening delimiter.
+* The content within the delimiters is treated as the documentation.
+* Leading whitespace followed by a single asterisk and a single following space on subsequent lines are considered stylistic and are stripped by doc generators.
```

Comment:
(What if there is a `*` that is not followed by a space? Is the `*` removed or not?)